docs/components/concepts/document-handling.md

```diff
@@ -5,56 +5,85 @@ description: "Learn more about integration, secure management, and efficient sto
+## Storage integration
 
-## Modeler
+### SaaS
 
-A [form](/components/modeler/forms/camunda-forms-reference.md) can display documents with the [document preview component](/components/modeler/forms/form-element-library/forms-element-library-document-preview.md):
+A [Google Cloud Platform](https://cloud.google.com/storage) storage integration is configured for SaaS and handled by Camunda.
 
-![document preview for form](./assets/document-preview.png)
+#### Limitations
+
+- One bucket per cluster is permitted with SaaS.
+- This storage integration is handled and configured by Camunda. While this is not dynamically configurable by the cluster, it is provided as environment configuration.
+- **Maximum file size**: 10 MB per file.
+- **File expiration time/time-to-live (TTL) policy**: 30 days by default. Clients for Connectors and Forms may specify a custom expiration date when uploading documents.
+
+### Self-Managed
+
+<!--- Does this content belong here, or should we link to it from the SM docs? --->
+
+Use [AWS S3](https://aws.amazon.com/s3/) storage and bucket creation per cluster to securely store and retrieve documents in an external, scalable storage solution, and to ensure storage is properly isolated and managed for each environment.
+
+#### Limitations
+
+<!--- same limitations as SaaS? if so, make one section --->
 
-With the [filepicker form element](/components/modeler/forms/form-element-library/forms-element-library-filepicker.md), you may also [upload multiple files at once](/components/modeler/forms/form-element-library/forms-element-library-filepicker.md#configurable-properties). This can be dynamically set using an [expression](/components/modeler/feel/language-guide/feel-expressions-introduction.md).
+<!--- insert section regarding API capabilities? --->
 
-## Connectors
+## Use cases and capabilities
 
-### Inbound (webhook) Connectors
+Document handling may be beneficial for several use cases. For example:
+
+- [Upload a document via the inbound webhook Connector](#upload-a-document-via-inbound-webhook-connector), and later retrieve the document content in another Connector invocation to store it in a third-party system.
+- [Upload a document via public form](#upload-a-document-via-public-form) to start the process, view and download files displayed in the task's form, and retrieve the document later in a Connector invocation to send it to a third-party system.
+
+### Upload a document via inbound webhook Connector
 
 [Access created documents](/components/connectors/protocol/http-webhook.md) in both the response expression and the result expression, where the `documents` object contains the references for created documents.
 
 ![example payload inbound webhook connector](./assets/inbound-webhook-document.png)
 
-### Outbound Connectors
+<!--- Explain how to then retrieve the document in another Connector to store it in a third-party system or provide a screenshot with some text --->
```

Comment:
@chillleader Can you help me understand this, or would you like me to add some time on your calendar to discuss?

Reply:
Shortly, the document reference that we receive as an output of one connector should be stored in process variables by using the result expression or result variable. Then, another connector can use this variable as an input. The format of inputs will depend on the connector, as each connector has a different input structure, but usually it would have a field called "Document(s)" and you would specify your variable with the document reference there.

Feel free to book something on my calendar if you need more info or a demo 👍